surfactant/relationships/dotnet_relationship.py

```diff
@@ -65,6 +66,58 @@ def establish_relationships(
                                 dnProbingPaths = []
                             dnProbingPaths.append(pathlib.PureWindowsPath(path).as_posix())
 
+    # https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/loading-unmanaged
+    # 1. Check the active AssemblyLoadContext cache
+    # 2. Calling the import resolver set by the setDllImportResolver function
+    #    - a. Example using SetDllImportResolver: https://learn.microsoft.com/en-us/dotnet/standard/native-interop/native-library-loading
+    #    - b. Checks PInvoke's or Assembly's DefaultDllImportSearchPathsAttribute, then the assembly's directory, then LoadLibraryEx with LOAD_WITH_ALTERED_SEARCH_PATH flag
+    #       - This attribute has no effect on non-Windows platforms / Mono runtime
+    #       - https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.defaultdllimportsearchpathsattribute?view=net-7.0
+    #       - i. This has a "Paths" property which is a bitwise combination of paths specified in ii:
+    #       - ii. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportsearchpath?view=net-7.0
+    # 3. The active AssemblyLoadContext calls its LoadUnmanagedDll function (Default behavior is the same as AssemblyRef probing?)
+    #    - a. Can be overridden, but the default implementation returns IntPtr.Zero, which tells the runtime to load with its default policy.
+    #    - b. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.loadunmanageddll?view=net-7.0
+    # 4. Run default unmanaged library probing logic by parsing *.deps.json probing properties
+    #    - a. If the json file isn't present, assume the calling assembly's directory contains the library
+    #    - b. https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing#unmanaged-native-library-probing
+    if "dotnetImplMap" in metadata:
+        for asmRef in metadata["dotnetImplMap"]:
+            if "Name" not in asmRef:
+                continue
+            refName = asmRef["Name"]
+            if is_absolute_path(refName):
+                relationships.extend(get_windows_pe_dependencies(sbom, software, [refName]))
```

Comment:
I don't think the `get_windows_pe_dependencies` function does what we want when a dependency is specified using absolute paths -- the function itself constructs a list of probe directories relative to the software, which when the given path is absolute we don't care about searching, and then it calls the `find_installed_software` function internally to find a match, which appends the file name for an import (which in this case would be an absolute path) to each probe directory given in turn.

For an example of a program in `C:/PrgFiles/Program.exe` an absolute import of `C:/SomeFolder/Dep.dll` looks like it would result in the function trying to search for something like `C:/PrgFiles/C:/Somefolder/Dep.dll`.

For matching an import listed as an absolute dependency, I think we can have a loop that just looks for a Software entry with an install path that just matches the absolute import name, without any of the probe directory things in it.

```python
ref_abspath = pathlib.PureWindowsPath(refName)
# iterate through all sbom entries
for e in sbom.software:
    # Skip if no install path (e.g. installer/temporary file)
    if e.installPath is None:
        continue
    if isinstance(e.installPath, Iterable):
        for ifile in e.installPath:
            # PureWindowsPath is case-insensitive for file/directory names
            if ref_abspath == pathlib.PureWindowsPath(ifile):
                # matching probe directory and filename, add relationship to list
                relationships.extend(Relationship(dependent_uuid, e.uuid, "Uses"))
```